In [116]:
import pandas as pd

#Data from: https://wwwn.cdc.gov/nchs/nhanes/continuousnhanes/default.aspx?BeginYear=2017

weighthistory_df = pd.read_sas("./data/WeightHistory_2017_18.xpt")
foodsecurity_df = pd.read_sas("./data/FoodSecurity_2017_18.xpt")
smokinghist_df = pd.read_sas("./data/SmokingHistory_CigUse_2017_18.xpt")
secondhandsmoke_df = pd.read_sas("./data/SmokingHistorySecondHand_2017_18.xpt")
rxdrugs_df = pd.read_sas("./data/PrescriptionDrugs_2017_18.xpt")
druguse_df = pd.read_sas("./data/DrugUse_2017_18.xpt")
nicotinemetabolites = pd.read_sas("./data/NicotineMetabolites_2017_18.xpt")
hdl_df = pd.read_sas("./data/CholesterolHDL_2017_18.xpt")
ldl_trig_df = pd.read_sas("./data/CholesterolLDL_2017_18.xpt")


nicotinebysmoke_df = smokinghist_df.merge(right=nicotinemetabolites, how="inner", on="SEQN")


print(druguse_df)


          SEQN  DUQ200  DUQ210  DUQ211  DUQ213  DUQ215Q  DUQ215U  DUQ217  \
0      93705.0     NaN     NaN     NaN     NaN      NaN      NaN     NaN   
1      93706.0     2.0     NaN     NaN     NaN      NaN      NaN     NaN   
2      93708.0     NaN     NaN     NaN     NaN      NaN      NaN     NaN   
3      93711.0     2.0     NaN     NaN     NaN      NaN      NaN     NaN   
4      93712.0     1.0    18.0     2.0     NaN      NaN      NaN     NaN   
...        ...     ...     ...     ...     ...      ...      ...     ...   
4567  102948.0     NaN     NaN     NaN     NaN      NaN      NaN     NaN   
4568  102949.0     1.0    18.0     1.0    18.0     15.0      1.0     3.0   
4569  102953.0     2.0     NaN     NaN     NaN      NaN      NaN     NaN   
4570  102954.0     2.0     NaN     NaN     NaN      NaN      NaN     NaN   
4571  102956.0     1.0    16.0     2.0     NaN      NaN      NaN     NaN   

      DUQ219  DUQ220Q  ...  DUQ380A  DUQ380B  DUQ380C  DUD380F  DUQ390  \
0        NaN 

Effects of stimulant use (or abuse) among prescribed and non-prescribed populations.

In [117]:
stimulants = adhd_medications = [
    "METHYLPHENIDATE",
    "DEXMETHYLPHENIDATE",
    "AMPHETAMINE",
    "DEXTROAMPHETAMINE",
    "LISDEXAMFETAMINE",
    "ATOMOXETINE",
    "GUANFACINE",
    "CLONIDINE"
]

antidepressants = [
    "FLUOXETINE",  # Prozac (SSRI)
    "SERTRALINE",  # Zoloft (SSRI)
    "ESCITALOPRAM",  # Lexapro (SSRI)
    "CITALOPRAM",  # Celexa (SSRI)
    "PAROXETINE",  # Paxil (SSRI)
    "FLUVOXAMINE",  # Luvox (SSRI)
    "VENLAFAXINE",  # Effexor (SNRI)
    "DESVENLAFAXINE",  # Pristiq (SNRI)
    "DULOXETINE",  # Cymbalta (SNRI)
    "LEVOMILNACIPRAN",  # Fetzima (SNRI)
    "BUPROPION",  # Wellbutrin (NDRI)
    "MIRTAZAPINE",  # Remeron (NaSSA)
    "VORTIOXETINE",  # Trintellix (Multimodal)
    "TRAZODONE",  # Desyrel (Serotonin Modulator)
    "NEFAZODONE",  # Serzone (Serotonin Modulator)
    "AMITRIPTYLINE",  # Elavil (TCA)
    "NORTRIPTYLINE",  # Pamelor (TCA)
    "IMIPRAMINE",  # Tofranil (TCA)
    "DESIPRAMINE",  # Norpramin (TCA)
    "DOXEPIN",  # Silenor (TCA)
    "SELEGILINE",  # Emsam (MAOI)
    "PHENELZINE",  # Nardil (MAOI)
    "TRANYLCYPROMINE",  # Parnate (MAOI)
    "ISOCARBOXAZID"  # Marplan (MAOI)
]

opioids = [
    "MORPHINE",
    "OXYCODONE",
    "HYDROCODONE",
    "CODEINE",
    "FENTANYL",
    "HYDROMORPHONE",
    "OXYMORPHONE",
    "METHADONE",
    "BUPEPRENORPHINE",
    "TRAMADOL",
    "MEPERIDINE",
    "TAPENTADOL",
    "LEVORPHANOL",
    "NALBUPHINE",
    "PENTAZOCINE"
]



rxdrugs_df["RXDDRUG"] = rxdrugs_df["RXDDRUG"].str.decode("utf-8")
#print(rxdrugs_df["RXDDRUG"].unique())

stimulants_regex = "|".join(stimulants)  # Creates a regex pattern like "METHYLPHENIDATE|DEXMETHYLPHENIDATE|..."
stimulants_df = rxdrugs_df[rxdrugs_df["RXDDRUG"].str.contains(stimulants_regex, regex=True, na=False)]

antidepressants_regex = "|".join(antidepressants)
antidepressants_df = rxdrugs_df[rxdrugs_df["RXDDRUG"].str.contains(antidepressants_regex, regex=True, na=False)]

opioids_regex = "|".join(opioids)
opioids_df = rxdrugs_df[rxdrugs_df["RXDDRUG"].str.contains(opioids_regex, regex=True, na=False)]
print(opioids_df.shape[0])



#print(stimulants_df.shape[0])
#print(filtered_df["RXDDRUG"].unique())
#print(rxdrugs_df[rxdrugs_df["RXDDRUG"].isin(stimulants)].shape[0])
#print(rxdrugs_df[rxdrugs_df["RXDDRUG"] == "METHYLPHENIDATE"].shape[0])


342


The output shows that in 2017, we have 849 unique individuals who were prescribed antidepressants, and 231 prescribed stimulants, annd 342 prescribed opiods. 

Nicotine Use Example:

Questionnaire data from : https://wwwn.cdc.gov/nchs/nhanes/search/datapage.aspx?Component=Questionnaire&CycleBeginYear=2017

*Nicotine by Smoking History variable names:*

Questionnaire:
* SMD030 - Age started smoking cigarettes regularly
* SMD641 - # days smoked cigs during past 30 days
* SMQ078 - How soon after waking do you smoke
* SMQ050Q - How long since quit smoking cigarettes

*Lab*
* URXCOTT - Cotinine, urine (ng/mL)
* URXNNCT - Nornicotine, urine (ng/mL)

In [118]:
#print(nicotinebysmoke_df[["SMD030", "SMD641", "URXCOTT", "URXNNCT"]])

nicotinebysmoke_df[["SMD030", "SMD641", "SMQ050Q", "SMQ078", "URXCOTT", "URXNNCT"]].corr()

SMD030    SMD641   SMQ050Q    SMQ078   URXCOTT   URXNNCT
SMD030   1.000000  0.025187  0.217885  0.002814 -0.034098 -0.035898
SMD641   0.025187  1.000000       NaN -0.071694  0.210681  0.241940
SMQ050Q  0.217885       NaN  1.000000       NaN  0.009650  0.025625
SMQ078   0.002814 -0.071694       NaN  1.000000 -0.085705 -0.090929
URXCOTT -0.034098  0.210681  0.009650 -0.085705  1.000000  0.840649
URXNNCT -0.035898  0.241940  0.025625 -0.090929  0.840649  1.000000

Number of days where one smoked in the past month were about 20% correlated with cotinine and nornicotine levels.